client/client_int_test.go

```diff
@@ -0,0 +1,202 @@
+		"prop2": "val2",
+	}
+
+	AfterEach(func() {
+		if a != nil {
+			a.Stop()
+		}
+	})
+
+	Describe("Async", func() {
+		startAppAndClient := func() {
+			var err error
+			a, err = app.NewApp("0.0.0.0", 5000, logger, config)
+			Expect(err).NotTo(HaveOccurred())
+			s = mocks.NewMockSender()
+			server := app.NewServer(s, logger)
+			a.SetServer(server)
+			go a.Run()
+			config.Set("client.async", true)
+			c, err = client.NewClient(
+				"",
+				config,
+				logger,
+				nil,
+				grpc.WithDefaultCallOptions(grpc.WaitForReady(true)),
+			)
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		Describe("Send", func() {
+			It("Should send event asynchronously", func() {
+				startAppAndClient()
+				err := c.Send(context.Background(), name, props)
+				Expect(err).NotTo(HaveOccurred())
+				c.Wait()
+				Expect(s.GetTotalSent()).To(Equal(1))
+			})
+
+			It("Should send multiple events asynchronously", func() {
+				startAppAndClient()
+				var err error
+				err = c.Send(context.Background(), name, props)
+				Expect(err).NotTo(HaveOccurred())
+				err = c.Send(context.Background(), name, props)
+				Expect(err).NotTo(HaveOccurred())
+				err = c.Send(context.Background(), name, props)
+				Expect(err).NotTo(HaveOccurred())
+				c.Wait()
+				Expect(s.GetTotalSent()).To(Equal(3))
+			})
+
+			Describe("Flush configurations", func() {
+				It("Should flush at every flushSize = 1 events", func() {
+					config.Set("client.flushInterval", 1*time.Second)
+					config.Set("client.flushSize", 1)
+					startAppAndClient()
+					var err error
+					err = c.Send(context.Background(), name, props)
+					Expect(err).NotTo(HaveOccurred())
+					err = c.Send(context.Background(), name, props)
+					Expect(err).NotTo(HaveOccurred())
+					err = c.Send(context.Background(), name, props)
+					Expect(err).NotTo(HaveOccurred())
+					c.Wait()
+					Expect(s.GetTotalSent()).To(Equal(3))
+					Expect(s.GetTotalCalls()).To(Equal(3))
+				})
+
+				It("Should flush at every flushSize = 3 events", func() {
+					config.Set("client.flushInterval", 1*time.Second)
+					config.Set("client.flushSize", 3)
+					startAppAndClient()
+					var err error
+					err = c.Send(context.Background(), name, props)
+					Expect(err).NotTo(HaveOccurred())
+					err = c.Send(context.Background(), name, props)
+					Expect(err).NotTo(HaveOccurred())
+					err = c.Send(context.Background(), name, props)
+					Expect(err).NotTo(HaveOccurred())
+					c.Wait()
+					Expect(s.GetTotalSent()).To(Equal(3))
+					Expect(s.GetTotalCalls()).To(Equal(1))
+				})
+
+				It("Should flush at every flushInterval = 5ms", func() {
+					config.Set("client.flushInterval", 5*time.Millisecond)
+					config.Set("client.flushSize", 3)
+					startAppAndClient()
+					var err error
+					err = c.Send(context.Background(), name, props)
+					Expect(err).NotTo(HaveOccurred())
+					err = c.Send(context.Background(), name, props)
+					Expect(err).NotTo(HaveOccurred())
+					time.Sleep(10 * time.Millisecond)
```

Comment:
Wait for 15+ millis, so we can test that it's not trying to send an empty buffer too?